src/infrasys/time_series_models.py

```diff
@@ -119,7 +119,7 @@ def aggregate(cls, ts_data: list[Self]) -> Self:
         InconsistentTimeseriesAggregation
             Raised if incompatible timeseries data are passed.
         """
-
+        agg_func = {"sum": sum, "avg": lambda x: sum(x) / len(x)}[agg_type]
```

Comment:
Peaky detail, but I am not a fam of defining functions inside functions. We can define this methods outside maybe?

Reply:
Also, wouldn't it be more consistent if we use numpy functions?